gtsam/geometry/Cal3.cpp

```diff
@@ -0,0 +1,72 @@
+/* ----------------------------------------------------------------------------
+
+ * GTSAM Copyright 2010, Georgia Tech Research Corporation,
+ * Atlanta, Georgia 30332-0415
+ * All Rights Reserved
+ * Authors: Frank Dellaert, et al. (see THANKS for the full author list)
+
+ * See LICENSE for the license information
+
+ * -------------------------------------------------------------------------- */
+
+/**
+ * @file   Cal3.cpp
+ * @brief  Common code for all calibration models.
+ * @author Frank Dellaert
+ */
+
+#include <gtsam/geometry/Cal3.h>
+
+#include <cmath>
+#include <fstream>
+#include <iostream>
+
+namespace gtsam {
+using namespace std;
+
+/* ************************************************************************* */
+Cal3::Cal3(double fov, int w, int h)
+    : s_(0), u0_((double)w / 2.0), v0_((double)h / 2.0) {
+  double a = fov * M_PI / 360.0;  // fov/2 in radians
+  fx_ =
+      (double)w / (2.0 * tan(a));  //    old formula: fx_ = (double) w * tan(a);
+  fy_ = fx_;
+}
+
+/* ************************************************************************* */
+Cal3::Cal3(const std::string& path)
+    : fx_(320), fy_(320), s_(0), u0_(320), v0_(140) {
+  char buffer[200];
+  buffer[0] = 0;
+  sprintf(buffer, "%s/calibration_info.txt", path.c_str());
```

Comment:
Security issue: potential out-of-buffer write. 
Why not?
```
const auto buffer = path  + std::string("/calibration_info.txt");
``` 
(would be shorter with c++14's [s-operator](https://en.cppreference.com/w/cpp/string/basic_string/operator%22%22s), but gtsam uses c++11 only).